packages/dev/gui/src/2D/controls/image.ts

```diff
@@ -525,19 +525,34 @@ export class Image extends Control {
         }
 
         // Should abstract platform instead of using LastCreatedEngine
-        const engine = this._host?.getScene()?.getEngine() || EngineStore.LastCreatedEngine;
-        if (!engine) {
-            throw new Error("Invalid engine. Unable to create a canvas.");
+
+        let fromCache, img;
+        if (this._host) {
+            const creationResult = this._host._getCanvasImage(value);
+            fromCache = creationResult.fromCache;
+            img = creationResult.img;
+        } else {
+            const engine = EngineStore.LastCreatedEngine;
+            if (!engine) {
+                throw new Error("Invalid engine. Unable to create a canvas.");
+            }
+            fromCache = false;
+            img = engine.createCanvasImage();
         }
-        this._domImage = engine.createCanvasImage();
 
-        this._domImage.onload = () => {
+        this._domImage = img;
+
+        if (fromCache) {
             this._onImageLoaded();
-        };
-        if (value) {
-            Tools.SetCorsBehavior(value, this._domImage);
-            Tools.SetReferrerPolicyBehavior(this.referrerPolicy, this._domImage);
-            this._domImage.src = value;
+        } else {
+            this._domImage.onload = () => {
+                this._onImageLoaded();

```

Comment:
Isn't this called twice in case of cache? Sorry if it isn't, might be the formatting of the changes.